stellar_sdk/asset.py

```diff
@@ -161,3 +161,8 @@ def __eq__(self, other: object) -> bool:
 
     def __str__(self):
         return f"<Asset [code={self.code}, issuer={self.issuer}, type={self.type}]>"
+    
+    def __hash__(self):
```

Comment:
The current `Asset` is not designed as an immutable object, if we add a __hash__ function like this, it may cause some unexpected behaviors, so I don’t plan to add it currently. I suggest users inherit it and implement their custom `Asset` class.

Reply:
Removed from the pull request.